grails-app/jobs/mama/ng/scheduler/SchedulerJob.groovy

```diff
@@ -0,0 +1,91 @@
+package mama.ng.scheduler
+
+import groovy.time.TimeCategory
+import groovyx.net.http.ContentEncoding
+import groovyx.net.http.ContentType
+import groovyx.net.http.HTTPBuilder
+import groovyx.net.http.HttpResponseException
+import groovyx.net.http.Method
+
+class SchedulerJob {
+    def concurrent = false
+
+    def cronParserService
+
+    static triggers = {
+        cron name:'cronTrigger', startDelay:1000, cronExpression: '0 0 * * * ?' } //execute every hour on the hour
+
+    def group = "Scheduler"
+
+    def execute() {
+        use (TimeCategory) {
+            def then = new Date() + 1.hour
+            def schedules = Schedule.findAllByNextSendLessThanEquals(then)
+
+            schedules.each { Schedule schedule ->
```

Comment:
I'm not quite following the logic at the moment. Every hour we seem to:
* Delete all messages for a schedule that is triggered.
* Create a new message object for each schedule that is triggered.
* Post an HTTP request for each schedule that is triggered.
* Then go through all messages with old nextSends and do HTTP posts for them.
I'm not sure what the last step is for because it seems like if ``message.nextSend <= then`` then ``schedule.nextSend <= then`` and the message will have been deleted before we can do anything with it in the last step?

For me the following would be more understandable:
* For each schedule, create a message with its current ``nextSend`` and then update the schedules ``nextSend``.
* For each message, do the HTTP post.
* Update the message with the results of the HTTP post.

Does Grails have a way to ensure that saving the schedule and creating the message happen inside the same transaction? That will prevent messages being lost if the process dies while dealing with schedules.